OSDOCS-9723: adds 4.15.5 relnotes Microshift (#73614)
openshift/openshift-docs: microshift_release_notes/microshift-4-15-release-notes.adoc

{product-title} release 4.15.0 is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHSA-2023:7200[RHSA-2023:7200] advisory. The images that are included in the update are provided by the link:https://access.redhat.com/errata/RHSA-2023:7198[RHSA-2023:7198] advisory.

> For the latest images included with {microshift-short}, view the contents of the `microshift-release-info` RPM. See xref:../microshift_install/microshift-embed-in-rpm-ostree-offline-use. adoc#microshift-embed-microshift-image-offline-deploy_microshift-embed-in-rpm-ostree-offline-use[Embedding {microshift-short} containers for offline deployments].

Reviewer: ```suggestion
For the latest images included with {microshift-short}, view the contents of the `microshift-release-info` RPM. See xref:../microshift_install/microshift-embed-in-rpm-ostree-offline-use.adoc#microshift-embed-microshift-image-offline-deploy_microshift-embed-in-rpm-ostree-offline-use[Embedding {microshift-short} containers for offline deployments].
```

Author: done

[id="microshift-4-15-2-dp"]
=== RHBA-2024:1212 - {microshift-short} 4.15.2 bug fix and enhancement update advisory

Issued: 2024-03-13

{product-title} release 4.15.2 is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHBA-2024:1212[RHBA-2024:1212] advisory. The images that are included in the update are provided by the link:https://access.redhat.com/errata/RHSA-2024:1210[RHSA-2024:1210] advisory.

> For the latest images included with {microshift-short}, view the contents of the `microshift-release-info` RPM. See xref:../microshift_install/microshift-embed-in-rpm-ostree-offline-use. adoc#microshift-embed-microshift-image-offline-deploy_microshift-embed-in-rpm-ostree-offline-use[Embedding {microshift-short} containers for offline deployments].

Reviewer: ```suggestion
For the latest images included with {microshift-short}, view the contents of the `microshift-release-info` RPM. See xref:../microshift_install/microshift-embed-in-rpm-ostree-offline-use.adoc#microshift-embed-microshift-image-offline-deploy_microshift-embed-in-rpm-ostree-offline-use[Embedding {microshift-short} containers for offline deployments].
```

Author: done

Issued: 2024-03-20

{product-title} release 4.15.3 is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHBA-2024:1257[RHBA-2024:1257] advisory. The images that are included in the update are provided by the link:https://access.redhat.com/errata/RHSA-2024:1255[RHSA-2024:1255] advisory.

For the latest images included with {microshift-short}, view the contents of the `microshift-release-info` RPM. See xref:../microshift_install/microshift-embed-in-rpm-ostree-offline-use. adoc#microshift-embed-microshift-image-offline-deploy_microshift-embed-in-rpm-ostree-offline-use[Embedding {microshift-short} containers for offline deployments].

[id="microshift-4-15-5-dp"]
=== RHBA-2024:1451 - {microshift-short} 4.15.5 bug fix and enhancement update advisory

Issued: 2024-03-27

{product-title} release 4.15.5 is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHBA-2024:1451[RHBA-2024:1451] advisory. The images that are included in the update are provided by the link:https://access.redhat.com/errata/RHSA-2024:1449[RHSA-2024:1449] advisory.

> 

Reviewer: ```suggestion

For the latest images included with {microshift-short}, view the contents of the `microshift-release-info` RPM. See xref:../microshift_install/microshift-embed-in-rpm-ostree-offline-use. adoc#microshift-embed-microshift-image-offline-deploy_microshift-embed-in-rpm-ostree-offline-use[Embedding {microshift-short} containers for offline deployments].
```

Author: done

[discrete]
[id="microshift-4-15-5-enhancements"]
==== Enhancements

Change in openshift-marketplace namespace security::

> * With this release, you should not set the `securityContextConfig` parameter when creating a catalog source in the `openshift-marketplace` namespace. However, you must set the `securityContextConfig` parameter to a value of `restricted` when creating a catalog source in other namespaces. (link:https://issues.redhat.com/browse/OCPBUGS-30034[OCPBUGS-30034]) (link:https://issues.redhat.com/browse/OCPBUGS-30034[OCPBUGS-30034])

Reviewer: ```suggestion
* With this release, the `openshift-marketplace` namespace security defaults to `baseline`. See the OLM documentation for specifics on how this change impacts your Operator deployments. See xref:../microshift_running_apps/microshift-operators-olm.adoc#microshift-operators-olm[Using Operator Lifecycle Manager with {microshift-short}]. (link:https://issues.redhat.com/browse/OCPBUGS-30034[OCPBUGS-30034]) 
```

Author: done

> For the latest images included with {microshift-short}, view the contents of the `microshift-release-info` RPM. See xref:../microshift_install/microshift-embed-in-rpm-ostree-offline-use. adoc#microshift-embed-microshift-image-offline-deploy_microshift-embed-in-rpm-ostree-offline-use[Embedding {microshift-short} containers for offline deployments].

Reviewer: ```suggestion

```

Author: done

Issued: 2024-03-20

{product-title} release 4.15.3 is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHBA-2024:1257[RHBA-2024:1257] advisory. The images that are included in the update are provided by the link:https://access.redhat.com/errata/RHSA-2024:1255[RHSA-2024:1255] advisory.

[id="microshift-4-15-4-dp"]
=== RHBA-2024:1451 - {microshift-short} 4.15.4 bug fix and enhancement update advisory

Issued: 2024-03-26

{product-title} release 4.15.4 is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHBA-2024:1451[RHBA-2024:1451] advisory. The images that are included in the update are provided by the link:https://access.redhat.com/errata/RHSA-2024:1449[RHSA-2024:1449] advisory.

[discrete]
[id="microshift-4-15-4-enhancements"]
==== Enhancements

> Change in OpenShift-marketplace namespace security::

Reviewer 2: @aravipra 
`OpenShift-marketplace` to `openshift-marketplace`

Reviewer 2: @aravipra you change OpenShift-marketplace to openShift-marketplace, but it should be openshift-marketplace (all are low cases)

please correct it, thanks

Author: done

> * In this release, the default value for the `securityContextConfig` parameter is changed from "restricted" to "baseline". (link:https://issues.redhat.com/browse/OCPBUGS-30034[OCPBUGS-30034])

Reviewer 2: @aravipra I do not think we use word `securityContextConfig` here because it could make the user to change 
```yaml
spec:
  grpcPodConfig:
    securityContextConfig: restricted
```
to
```yaml
spec:
  grpcPodConfig:
    securityContextConfig: baseline
```

but actually no.

after changing `pod-security.kubernetes.io/enforce: restricted` to `pod-security.kubernetes.io/enforce: baseline` for openshift-marketplace in microshift with https://issues.redhat.com/browse/OCPBUGS-30034, when creating catalog source in openshift-marketplace, could keep
```yaml
spec:
  grpcPodConfig:
    securityContextConfig: restricted
```
or remove this part to take default value legacy.

if catalog source is create in other ns which enforce is restricted, still keep it.
```yaml
spec:
  grpcPodConfig:
    securityContextConfig: restricted
```

so, with https://issues.redhat.com/browse/OCPBUGS-30034, I suggest to change the word to
```console
In this release, if you create catalog source in openshift-marketplace, do not need to set securityContextConfig. if you create catalog source in other ns, still need to set securityContextConfig to restricted
```

by the way, securityContextConfig in catalogsource only support restricted or legacy.

cc @kevinrizza @jianzhangbjz @pmtk

Author: done

> * In this release, the `securityContextConfig` parameter is not to be set when creating a catalog source in the OpenShift-marketplace namespace. However, the `securityContextConfig` parameter must be "restricted" when creating a catalog source in other namespaces. (link:https://issues.redhat.com/browse/OCPBUGS-30034[OCPBUGS-30034])

Reviewer 2: @aravipra same above

Author: done

Reviewer 2: continue to correct it. thanks

> * In this release, the `securityContextConfig` parameter is not to be set when creating a catalog source in the openshift-marketplace namespace. However, the `securityContextConfig` parameter must be "restricted" when creating a catalog source in other namespaces. (link:https://issues.redhat.com/browse/OCPBUGS-30034[OCPBUGS-30034])

Author: QE has confirmed that this is a enhancement. 
CC: @kuiwang02

> * In this release, the `securityContextConfig` parameter is not to be set when creating a catalog source in the openshift-marketplace namespace. However, the `securityContextConfig` parameter must be "restricted" when creating a catalog source in other namespaces. (link:https://issues.redhat.com/browse/OCPBUGS-30034[OCPBUGS-30034])

Reviewer 3: I guess my suggestion then would be to work with QE to write this up so that it sounds more like an enhancement rather than a bug or KI.

> * In this release, the `securityContextConfig` parameter is not to be set when creating a catalog source in the openshift-marketplace namespace. However, the `securityContextConfig` parameter must be "restricted" when creating a catalog source in other namespaces. (link:https://issues.redhat.com/browse/OCPBUGS-30034[OCPBUGS-30034])

Reviewer: I suggest 

* With this release, the `securityContextConfig` parameter no longer needs to be changed from the default value of baseline when creating a catalog source in the openshift-marketplace namespace. However, the `securityContextConfig` parameter must be set to restricted when creating a catalog source in other namespaces. (link:https://issues.redhat.com/browse/OCPBUGS-30034[OCPBUGS-30034]) (edited) 

But my question is, does this change impact only the CS used with oc mirror, or all catalog sources we're using with OLM? @pmtk

> * In this release, the `securityContextConfig` parameter is not to be set when creating a catalog source in the openshift-marketplace namespace. However, the `securityContextConfig` parameter must be "restricted" when creating a catalog source in other namespaces. (link:https://issues.redhat.com/browse/OCPBUGS-30034[OCPBUGS-30034])

Reviewer 2: @aravipra 
```console
In this release, you should not set the `securityContextConfig` parameter when creating a catalog source in the openshift-marketplace namespace. However, you must set the `securityContextConfig` parameter to a value of `restricted` when creating a catalog source in other namespaces. (link:https://issues.redhat.com/browse/OCPBUGS-30034[OCPBUGS-30034])
```
from @JoeAldinger is ok. it is same with mine with better word.

@ShaunaDiaz 
`securityContextConfig` has nothing with baseline whether there is this fix or not. baseline work for enforce of namespace which impact catalogsoruce pod creation. but `securityContextConfig`'s value has no baseline.
so your word is not correct.

and more, the fix impact all cs creation what ever it is cs used with oc-miorror or not.

> * In this release, the `securityContextConfig` parameter is not to be set when creating a catalog source in the openshift-marketplace namespace. However, the `securityContextConfig` parameter must be "restricted" when creating a catalog source in other namespaces. (link:https://issues.redhat.com/browse/OCPBUGS-30034[OCPBUGS-30034])

Author: @JoeAldinger I do not see the use of  word "you" in the Microshift release notes. Also the word "should" has to be avoided.

> * In this release, the `securityContextConfig` parameter is not to be set when creating a catalog source in the openshift-marketplace namespace. However, the `securityContextConfig` parameter must be "restricted" when creating a catalog source in other namespaces. (link:https://issues.redhat.com/browse/OCPBUGS-30034[OCPBUGS-30034])

Reviewer 4: I agree with @kuiwang02 on
> and more, the fix impact all cs creation what ever it is cs used with oc-miorror or not

For the rest, I'm bit confused - I'm not an expert in that domain so I can only say what I saw and what worked.

When `openshift-marketplace` had security `restricted`: CatalogSource generated by oc-mirror doesn't have `securityContextConfig` so the Pod with the catalog wasn't starting. This could be remedied by addition of `grpcPodConfig.securityContextConfig: restricted` or editing the `openshift-marketplace` ns on the cluster.

After we changed `openshift-marketplace` namespace's security to `baseline`, the CatalogSource from oc-mirror (still without securityContextConfig) worked without modifying it.

I don't know the exact implications and iteration between these two values so I'll let OLM experts decide.

> * In this release, the `securityContextConfig` parameter is not to be set when creating a catalog source in the openshift-marketplace namespace. However, the `securityContextConfig` parameter must be "restricted" when creating a catalog source in other namespaces. (link:https://issues.redhat.com/browse/OCPBUGS-30034[OCPBUGS-30034])

Author: done

Issued: 2024-03-20

{product-title} release 4.15.3 is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHBA-2024:1257[RHBA-2024:1257] advisory. The images that are included in the update are provided by the link:https://access.redhat.com/errata/RHSA-2024:1255[RHSA-2024:1255] advisory.

> For the latest images included with {microshift-short}, view the contents of the `microshift-release-info` RPM. See xref:../microshift_install/microshift-embed-in-rpm-ostree-offline-use. adoc#microshift-embed-microshift-image-offline-deploy_microshift-embed-in-rpm-ostree-offline-use[Embedding {microshift-short} containers for offline deployments].

Author: done

[id="microshift-4-15-5-dp"]
=== RHBA-2024:1451 - {microshift-short} 4.15.5 bug fix and enhancement update advisory

Issued: 2024-03-27

{product-title} release 4.15.5 is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHBA-2024:1451[RHBA-2024:1451] advisory. The images that are included in the update are provided by the link:https://access.redhat.com/errata/RHSA-2024:1449[RHSA-2024:1449] advisory.

> For the latest images included with {microshift-short}, view the contents of the `microshift-release-info` RPM. See xref:../microshift_install/microshift-embed-in-rpm-ostree-offline-use. adoc#microshift-embed-microshift-image-offline-deploy_microshift-embed-in-rpm-ostree-offline-use[Embedding {microshift-short} containers for offline deployments].

Reviewer: ```suggestion
For the latest images included with {microshift-short}, view the contents of the `microshift-release-info` RPM. See xref:../microshift_install/microshift-embed-in-rpm-ostree-offline-use.adoc#microshift-embed-microshift-image-offline-deploy_microshift-embed-in-rpm-ostree-offline-use[Embedding {microshift-short} containers for offline deployments].
```
My mistake, sorry about that!

Author: done. NP!
